servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -444,8 +445,12 @@ public void handle(final HttpServiceContext ctx, final BlockingStreamingHttpRequ
                             final DefaultGrpcPayloadWriter<Resp> grpcPayloadWriter =
                                     new DefaultGrpcPayloadWriter<>(response.sendMetaData(serializer));
                             try {
+                                // Set status OK before invoking handle methods because users can close PayloadWriter
+                                final HttpPayloadWriter<Resp> payloadWriter = grpcPayloadWriter.payloadWriter();
+                                setStatusOk(payloadWriter.trailers(), ctx.executionContext().bufferAllocator());
```

Comment:
Did you consider setting a status in `grpcPayloadWriter.close()` if it is not set?

Reply:
Yes, I considered that approach, but then we have to store the ref for `BufferAllocator` inside `DefaultGrpcPayloadWriter` and do a look up for the header before setting `OK`.
Because `OK` is the most common case and we do not expose trailers in `handle` method (users can't change them), this approach is good enough.

Reply:
>> this approach is good enough.

Agreed, thanks for the explanation